pol-core/bscript/compiler/codegen/FunctionReferenceRegistrar.h

```diff
@@ -0,0 +1,40 @@
+#pragma once
+
+#include <map>
+#include <memory>
+#include <string>
+#include <vector>
+
+namespace Pol::Bscript::Compiler
+{
+class FlowControlLabel;
+class UserFunction;
+class FunctionReferenceDescriptor;
+
+class FunctionReferenceRegistrar
+{
+public:
+  FunctionReferenceRegistrar();
+
+
+  std::vector<FunctionReferenceDescriptor> take_descriptors();
+
+  void lookup_or_register_reference( const UserFunction& node, unsigned& reference_index );
```

Comment:
Returning the reference_index would make the calling side clearer

Reply:
I wanted to keep with the existing prototype style: https://github.com/polserver/polserver/blob/e8b5c15ca7bdb3dff22ca66795e71413b21b6adf/pol-core/bscript/compiler/codegen/ModuleDeclarationRegistrar.h#L25-L26